tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1560,6 +1568,48 @@ bool StorageManager::is_vacuum_file(const URI& uri) const {
   return false;
 }
 
+Status StorageManager::get_array_schema_uris(
+    const URI& array_uri, std::vector<URI>* uris) const {
+  STATS_START_TIMER(stats::GlobalStats::TimerType::READ_GET_ARRAY_SCHEMAS)
+
+  const URI schema_uri =
+      array_uri.join_path(constants::array_schema_folder_name);
+  bool is_dir = false;
+  RETURN_NOT_OK(vfs_->is_dir(schema_uri, &is_dir));
+  if (!is_dir) {
+    // If it is not a directory, the array schema could be stored in the
+    // previous format
+    uris->clear();
+    return Status::Ok();
+  }
+  RETURN_NOT_OK(vfs_->ls(schema_uri, uris));
+  // Sort schema URIs
+  std::sort(uris->begin(), uris->end());
```

Comment:
Not sure if this is needed - is `ls` guaranteed to return sorted filenames?

Reply:
you are correct. Not necessary to sort it again. "ls" is guaranteed to return for new added filenames.